src/cdomains/intDomain.ml

```diff
@@ -616,9 +617,8 @@ struct
     | `Bot
   ] [@@deriving to_yojson]
   let name () = "def_exc"
-  let top_of ik = `Excluded (S.empty (), size ik)
-  let top_size = Size.max `Signed
-  let top () = top_of top_size
+  let top_range = R.of_interval (-99L, 99L) (* Since there is no top ikind we use a range that includes both ILongLong [-63,63] and IULongLong [0,64]. Only needed for intermediate range computation on longs. Correct range is set by cast. *)
+  let top () = `Excluded (S.empty (), top_range)
```

Comment:
Since we call `ID.top ()` in quite some places, values of range [-99,99] have the potential to propagate to a lot of places. Maybe it would be good to once again to the `cast_to ik` wherever we know the `ik` for an `ID.top ()` to limit this.

Reply:
Yes, it can be seen as a poison value that should never make its way into results (at least for C integers).
This PR is based on #93, so it should still contain those cast calls.

Reply:
I was comparing against #89 and there is a few more casts there, not sure which ones (if any) one would like to also have here:
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix

Reply:
Apparently none of them are needed. All those casts in base make the code pretty hard to read.

Reply:
Some things that are actually different:

base:
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R230
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R691
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R1344
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R1417-R1418
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R1482-R1483
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-a9a7366a202e75b5e023ecd14b20ce90R2050

valueDomain:
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-d1d6b7a9c4034aaf9a2f822ca2cf0b02R403
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-d1d6b7a9c4034aaf9a2f822ca2cf0b02R431
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-d1d6b7a9c4034aaf9a2f822ca2cf0b02R459
https://github.com/goblint/analyzer/compare/fix/ranges..def_exc_fix#diff-d1d6b7a9c4034aaf9a2f822ca2cf0b02R524

Reply:
Let's go through those in https://github.com/goblint/analyzer/pull/93#issuecomment-686506489 since they're changes in the parent of this branch. Probably accept most of them?